From 0c9c7408537a543b10035434d79842651f540d7d Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Sun, 16 Nov 2014 17:09:32 +0100
Subject: [PATCH 12/15] ARM: dts: sunxi: Add de_be0 clk parent pll to simplefb
 node
Origin: https://git.kernel.org/linus/678e75d3e5526bee1b09f0e916096ec7f7d35c8e

Avoid the parent pll for the mod-clk for de_be0 getting disabled when non of
the other users are enabled (which can happen when none of i2c, spi and mmc
are in use).

Note for now we point directly to the parent rather then to the de_be0 mod-clk
as that is not modelled in our devicetree yet.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
---
 arch/arm/boot/dts/sun4i-a10.dtsi  | 3 ++-
 arch/arm/boot/dts/sun5i-a10s.dtsi | 3 ++-
 arch/arm/boot/dts/sun6i-a31.dtsi  | 1 +
 arch/arm/boot/dts/sun7i-a20.dtsi  | 3 ++-
 4 files changed, 7 insertions(+), 3 deletions(-)

Index: tree-jessie/arch/arm/boot/dts/sun4i-a10.dtsi
===================================================================
--- tree-jessie.orig/arch/arm/boot/dts/sun4i-a10.dtsi
+++ tree-jessie/arch/arm/boot/dts/sun4i-a10.dtsi
@@ -34,7 +34,8 @@
 
 		framebuffer0-hdmi {
 			compatible = "simple-framebuffer";
-			clocks = <&ahb_gates 36>, <&ahb_gates 43>, <&ahb_gates 44>;
+			clocks = <&pll5 1>, <&ahb_gates 36>, <&ahb_gates 43>,
+				 <&ahb_gates 44>;
 			status = "disabled";
 		};
 	};
Index: tree-jessie/arch/arm/boot/dts/sun5i-a10s.dtsi
===================================================================
--- tree-jessie.orig/arch/arm/boot/dts/sun5i-a10s.dtsi
+++ tree-jessie/arch/arm/boot/dts/sun5i-a10s.dtsi
@@ -31,7 +31,8 @@
 
 		framebuffer0-hdmi {
 			compatible = "simple-framebuffer";
-			clocks = <&ahb_gates 36>, <&ahb_gates 43>, <&ahb_gates 44>;
+			clocks = <&pll5 1>, <&ahb_gates 36>, <&ahb_gates 43>,
+				 <&ahb_gates 44>;
 			status = "disabled";
 		};
 	};
Index: tree-jessie/arch/arm/boot/dts/sun6i-a31.dtsi
===================================================================
--- tree-jessie.orig/arch/arm/boot/dts/sun6i-a31.dtsi
+++ tree-jessie/arch/arm/boot/dts/sun6i-a31.dtsi
@@ -32,6 +32,7 @@
 
 		framebuffer0-hdmi {
 			compatible = "simple-framebuffer";
+			clocks = <&pll6 0>;
 			status = "disabled";
 		};
 	};
Index: tree-jessie/arch/arm/boot/dts/sun7i-a20.dtsi
===================================================================
--- tree-jessie.orig/arch/arm/boot/dts/sun7i-a20.dtsi
+++ tree-jessie/arch/arm/boot/dts/sun7i-a20.dtsi
@@ -35,7 +35,8 @@
 
 		framebuffer0-hdmi {
 			compatible = "simple-framebuffer";
-			clocks = <&ahb_gates 36>, <&ahb_gates 43>, <&ahb_gates 44>;
+			clocks = <&pll5 1>, <&ahb_gates 36>, <&ahb_gates 43>,
+				 <&ahb_gates 44>;
 			status = "disabled";
 		};
 	};
